### PR TITLE
170 - System Search and Station Search By System Name for Trade

### LINKED ISSUE #170: System-Search_Station-Search-By-System
Add two endpoints, find System by substring to match exploration, find Stations by SystemName to return a station list.

### PR DESCRIPTION
**Pull request by Daniel-J-Mason**
closes #170

### CODE REVIEW

File: trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisSystemRepository.java [hunk 1]
```diff
@@ -75,4 +75,14 @@ SELECT id, name, elite_id, ST_X(coordinates_geom) as x_coordinate, ST_Y(coordina
...
+    
+    @Select({
+            "SELECT id, name, elite_id, ST_X(coordinates_geom) as x_coordinate, ST_Y(coordinates_geom) as y_coordinate, ST_Z(coordinates_geom) as z_coordinate " +
+                    "FROM system " +
+                    "WHERE name ILIKE CONCAT('%', #{name}, '%')" +
+                    "ORDER BY CASE WHEN name ILIKE CONCAT(#{name}, '%') THEN 0 ELSE 1 END, name " +
+                    "LIMIT #{amount}"
+    })
+    @ResultMap("systemResultMap")
+    List<MybatisSystemEntity> findSystemsByNameContaining(@Param("name") String name, @Param("amount") Integer amount);
```
Comment (aslink87) [inline trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisSystemRepository.java:87]: Could you performance test this query as is vs w/o a leading wildcard? I don't think it's appropriate to search for systems by name containing. Instead would it not make more sense to search by name beginning with? Open to discussion here. For reference, using Insomnia to query against `https://duval.edpn.io/api/v1/exploration/system/by-name-containing?subString=Sol&amount=10` takes me from 600ms to 1s.

Comment (Daniel-J-Mason) [reply trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisSystemRepository.java:87]: go ahead and hit that endpoint again, ill push a separate commit to add it properly, but I just threw a temporary index on it that brings query time down to <10ms so its all api time to present now. Let me know if this is fast enough, I can curl it in half a second.

I think its relevant to do name containing. sometimes you can only remember the second word in a station name, list comes in alphabetized, with containing results at the tail end.

Comment (aslink87) [reply trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisSystemRepository.java:87]: now getting in the 130ms range. much better. thanks

**Comment by Daniel-J-Mason** [comment]
Closed to be revisited in the future.

Going for one endpoint for each search bar/ drop down

**Pull request closed without merging**